packages/business-rules-lib/src/validators/contact.js

```diff
@@ -286,3 +286,25 @@ export const createLastNameValidator = joi =>
     .external(capitaliseNamePrefixes(['Mc', "O'"]))
     .required()
     .example('Tester')
+
+/**
+ * Create a validator to check a valid national insurance number
+ *
+ * @param {Joi.Root} joi the joi validator used by the consuming project
+ * @returns {Joi.AnySchema}
+ */
+
+const ukNINORegEx = /^([ABCEGHJ-PRSTW-Z][ABCEGHJ-NPRSTW-Z])\s?([0-9]{2})\s{0,3}([0-9]{2})\s{0,3}([0-9]{2})\s{0,3}([ABCD])$/
```

Comment:
These regular expressions should be combined into a single expression for efficiency.  A negative lookbehind can be used to accomplish this.  I haven't tested it but something like this should work:

`const ukNINORegEx = /^([ABCEGHJ-PRSTW-Z][ABCEGHJ-NPRSTW-Z])(?<!(?BG|GB|KN|NK|NT|TN|ZZ))\s?([0-9]{2})\s{0,3}([0-9]{2})\s{0,3}([0-9]{2})\s{0,3}([ABCD])$/`

Reply:
Done